sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -327,6 +327,14 @@ object TypeCoercion {
       // Skip nodes who's children have not been resolved yet.
       case e if !e.childrenResolved => e
 
+      // For integralType should not convert to double which will cause precision loss.
+      case a @ BinaryArithmetic(left @ StringType(), right @ IntegralType()) =>
```

Comment:
What will happen if string value beyond the long type range?

Reply:
Thanks @wangyum , it will return `NULL`.
I modify to use `DecimalType.SYSTEM_DEFAULT` instead. 
I consider to check value, but i think `DecimalType.SYSTEM_DEFAULT` is enough.What do you think about this?

Reply:
@wangyum Sorry for bothering you, i will take some time to fix this later .